RecoParticleFlow/PFProducer/plugins/importers/GeneralTracksImporter.cc

```diff
@@ -13,33 +14,84 @@ class GeneralTracksImporter : public BlockElementImporterBase {
-        DPtovPtCut_(conf.getParameter<std::vector<double> >("DPtOverPtCuts_byTrackAlgo")),
-        NHitCut_(conf.getParameter<std::vector<unsigned> >("NHitCuts_byTrackAlgo")),
+        DPtovPtCut_(conf.getParameter<std::vector<double>>("DPtOverPtCuts_byTrackAlgo")),
+        NHitCut_(conf.getParameter<std::vector<unsigned>>("NHitCuts_byTrackAlgo")),
         useIterTracking_(conf.getParameter<bool>("useIterativeTracking")),
         cleanBadConvBrems_(conf.getParameter<bool>("cleanBadConvertedBrems")),
-        muonMaxDPtOPt_(conf.getParameter<double>("muonMaxDPtOPt")) {}
+        muonMaxDPtOPt_(conf.getParameter<double>("muonMaxDPtOPt")) {
+    if (vetoEndcap_) {
+      vetoMode_ = conf.getParameter<unsigned>("vetoMode");
+      switch (vetoMode_) {
+        case pfRecTrackCollection:
+          vetoPFTracksSrc_ = sumes.consumes<reco::PFRecTrackCollection>(conf.getParameter<edm::InputTag>("vetoSrc"));
+          break;
+        case ticlSeedingRegion:
+          vetoTICLSeedingSrc_ =
+              sumes.consumes<std::vector<TICLSeedingRegion>>(conf.getParameter<edm::InputTag>("vetoSrc"));
+          tracksSrc_ = sumes.consumes<reco::TrackCollection>(conf.getParameter<edm::InputTag>("tracksSrc"));
+          break;
+        case pfCandidateCollection:
+          vetoPFCandidatesSrc_ =
+              sumes.consumes<reco::PFCandidateCollection>(conf.getParameter<edm::InputTag>("vetoSrc"));
+          break;
+        default:;
+      }  // switch
+    }
+  }
 
   void importToBlock(const edm::Event&, ElementList&) const override;
 
 private:
-  int muAssocToTrack(const reco::TrackRef& trackref, const edm::Handle<reco::MuonCollection>& muonh) const;
-
-  edm::EDGetTokenT<reco::PFRecTrackCollection> src_;
-  edm::EDGetTokenT<reco::MuonCollection> muons_;
+  const edm::EDGetTokenT<reco::PFRecTrackCollection> src_;
+  const bool vetoEndcap_;
+  const edm::EDGetTokenT<reco::MuonCollection> muons_;
   const reco::TrackBase::TrackQuality trackQuality_;
   const std::vector<double> DPtovPtCut_;
   const std::vector<unsigned> NHitCut_;
   const bool useIterTracking_, cleanBadConvBrems_;
   const double muonMaxDPtOPt_;
+  unsigned int vetoMode_;
+  edm::EDGetTokenT<reco::PFRecTrackCollection> vetoPFTracksSrc_;
+  edm::EDGetTokenT<std::vector<TICLSeedingRegion>> vetoTICLSeedingSrc_;
+  edm::EDGetTokenT<reco::TrackCollection> tracksSrc_;
+  edm::EDGetTokenT<reco::PFCandidateCollection> vetoPFCandidatesSrc_;
 };
 
 DEFINE_EDM_PLUGIN(BlockElementImporterFactory, GeneralTracksImporter, "GeneralTracksImporter");
 
 void GeneralTracksImporter::importToBlock(const edm::Event& e, BlockElementImporterBase::ElementList& elems) const {
   typedef BlockElementImporterBase::ElementList::value_type ElementType;
   auto tracks = e.getHandle(src_);
+
+  typedef std::pair<edm::ProductID, unsigned> TrackProdIDKey;
+  std::vector<TrackProdIDKey> vetoed;
+  if (vetoEndcap_) {
+    if (vetoMode_ == pfRecTrackCollection) {
```

Comment:
a switch is more appropriate here so that we even get help from the compiler that all cases are covered